src/common/id/SegmentId.h

```diff
@@ -0,0 +1,81 @@
+/* Copyright (c) 2021 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License.
+ */
+
+#ifndef COMMON_ID_SEGMENTINCR_H_
+#define COMMON_ID_SEGMENTINCR_H_
+
+#include <double-conversion/utils.h>
+
+#include "clients/meta/MetaClient.h"
+#include "common/base/Base.h"
+
+namespace nebula {
+// Segment auto-increase id
+class SegmentId {
+  FRIEND_TEST(SegmentIdTest, TestConcurrency);
+
+ public:
+  static SegmentId& getInstance() {
+    ASSERT(client_ != nullptr);
+    ASSERT(runner_ != nullptr);
+    static SegmentId instance;
+    return instance;
+  }
+
+  ~SegmentId() = default;
+
+  SegmentId(const SegmentId&) = delete;
+
+  SegmentId& operator=(const SegmentId&) = delete;
+
+  Status init(int64_t step) {
```

Comment:
I don't see any usage of this interface in our main process. Where do you plan to use it? By the way, the step should be configurable.